src/components/recipes/Ingredient.tsx

```diff
@@ -81,9 +76,15 @@ const Ingredient: FC<Props> = ({ ingredient }) => {
       });
 
       editName(ingredient.name);
+      editSortBy(ingredient.sortBy);
+      editOptimize(ingredient.optimize);
 
-      //if not /produkty, it's the ingredient inside createRecipe and I want to add amount
-      location.pathname === "/produkty" ? onEditOpen() : onEditInRecipeOpen();
+      if ("amount" in ingredient) {
```

Comment:
Proc ne `ingredient.amount`? `in` operator je trochu nebezpecnej protoze kouka na vsechny "properties" na prototypu. `"toString" in ingredient` = true

Reply:
v props predavam dva typy ingredienci, jeden nema amount

Reply:
Na to muzem pozuit Type Guard https://blog.logrocket.com/how-to-use-type-guards-typescript/